File: algorithm-pattern-problems/c/array_and_hashing/contains_duplicate.c
#include <stdlib.h>
#include <stdbool.h>

int compare(const void *a, const void *b){
	return *(int*)a - *(int*)b;
}

bool containsDuplicate(int* nums, int numsSize) {
   // sort the array
   qsort(nums, numsSize, sizeof(int), compare);

   // check for adjacent duplicate element
   for (int i = 1; i < numsSize; i++)
   {
	  if (nums[i] == nums[i-1])
	  {
		  return true;
	  }
   }

   return false;
   // Time: O(n logn)
   // space: O(1)
}

File: algorithm-pattern-problems/golang/array_and_hashing/contains_duplicate.go
package arrayandhashing

func containsDuplicate(nums []int) bool {
	if len(nums) == 1 {
		return false
	}

	hash := make(map[int]int, 0)

	for _, num := range nums {
		if _, ok := hash[num]; ok {
			return true
		}

		hash[num]++
	}

	return false
	// Time: O(n)
	// Space: O(n)
}
